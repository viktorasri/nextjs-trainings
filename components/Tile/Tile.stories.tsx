import { expect } from '@storybook/jest';
import { ComponentMeta, ComponentStoryObj } from '@storybook/react';
import { screen } from '@storybook/testing-library';
import { Tile } from './Tile';

export default {
  title: 'Content/Tile',
  component: Tile,
} as ComponentMeta<typeof Tile>;

export const BasicTile: ComponentStoryObj<typeof Tile> = {
  play: async () => {
    await expect(screen.getByRole('heading')).toBeInTheDocument();
  },
  args: {
    children: '{{children}}',
    header: 'Lorem ipsum dolor sit amet',
  },
};
