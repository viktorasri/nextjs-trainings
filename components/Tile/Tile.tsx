import styled from '@emotion/styled';
import { FC, PropsWithChildren } from 'react';
import { borderRadius, boxShadow } from '../styles';

interface ITile extends PropsWithChildren {
  header: string;
}

const Section = styled.section`
  padding: 1vmin 4vmin 4vmin;
  background: ${({ theme }) => theme.background};
  color: ${({ theme }) => theme.font.regular};
  ${borderRadius()}
  ${({ theme }) =>
    boxShadow(theme.components.shadow1, theme.components.shadow1)}
`;

export const Tile: FC<ITile> = ({ header, children }) => {
  return (
    <Section>
      <h2>{header}</h2>
      {children}
    </Section>
  );
};
