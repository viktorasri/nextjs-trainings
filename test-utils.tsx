import { ThemeProvider } from '@emotion/react';
import { render, RenderOptions } from '@testing-library/react';
import { FC, PropsWithChildren, ReactElement } from 'react';
import { Themes } from './styles/themes';

const Wrapper: FC<PropsWithChildren> = ({ children }) => (
  <ThemeProvider theme={Themes.light}>{children}</ThemeProvider>
);

const customRender = (ui: ReactElement, options?: RenderOptions) => {
  return render(ui, { wrapper: Wrapper, ...options });
};

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
