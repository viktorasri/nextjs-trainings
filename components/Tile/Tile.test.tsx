import { render } from '~/test-utils';
import { Tile } from './Tile';

describe('Tile test cases', () => {
  it('Tile render check', () => {
    const { asFragment } = render(
      <Tile header="Lorem ipsum dolor sit amet">
        <p>
          Etiam pulvinar ornare ornare. Vestibulum eget ante eget elit fringilla
          ultricies. Maecenas pretium mauris non porttitor semper. Nullam non
          interdum ex. Proin vel justo bibendum, iaculis justo sit amet,
          ullamcorper sem. Donec vitae velit in quam posuere accumsan ac ut est.
          Ut suscip.
        </p>
      </Tile>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
