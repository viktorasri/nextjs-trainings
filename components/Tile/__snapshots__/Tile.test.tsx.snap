// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile test cases Tile render check 1`] = `
<DocumentFragment>
  <section
    class="css-qlxy6r"
  >
    <h2>
      Lorem ipsum dolor sit amet
    </h2>
    <p>
      Etiam pulvinar ornare ornare. Vestibulum eget ante eget elit fringilla ultricies. Maecenas pretium mauris non porttitor semper. Nullam non interdum ex. Proin vel justo bibendum, iaculis justo sit amet, ullamcorper sem. Donec vitae velit in quam posuere accumsan ac ut est. Ut suscip.
    </p>
  </section>
</DocumentFragment>
`;
